Libtool 2.2.2 can't cope unless the LT_ variables are marked as AC_SUBST
as otherwise autofoo tries to expand them, fails and gets upset.

You can't conditionally add AC_PROG_CXX to a configure.ac file since the
mere mention of it triggers some internals which then get upset it if
wasn't really called at configure time.

Referring to a .cpp file in Makefiles without calling AC_PROG_CXX is
now also a fatal error. Easiest fix is to drop the CXX stuff entirely.

RP - 14/4/08

Index: cairo-1.4.14/configure.in
===================================================================
--- cairo-1.4.14.orig/configure.in	2008-04-14 14:50:16.000000000 +0100
+++ cairo-1.4.14/configure.in	2008-04-14 15:05:59.000000000 +0100
@@ -23,15 +23,18 @@
 
 # Increment if the interface has additions, changes, removals.
 LT_CURRENT=13
+AC_SUBST(LT_CURRENT)
 
 # Increment any time the source changes; set to
 # 0 if you increment CURRENT
 LT_REVISION=7
+AC_SUBST(LT_REVISION)
 
 # Increment if any interfaces have been added; set to 0
 # if any interfaces have been removed. removal has
 # precedence over adding, so set to 0 if both happened.
 LT_AGE=11
+AC_SUBST(LT_AGE)
 
 dnl ===========================================================================
 
@@ -342,9 +345,6 @@
   esac
 ])
 
-if test "x$use_beos" = "xyes"; then
-  AC_PROG_CXX
-fi
 
 dnl ===========================================================================
 
Index: cairo-1.4.14/boilerplate/Makefile.am
===================================================================
--- cairo-1.4.14.orig/boilerplate/Makefile.am	2008-04-14 15:08:47.000000000 +0100
+++ cairo-1.4.14/boilerplate/Makefile.am	2008-04-14 15:08:57.000000000 +0100
@@ -9,7 +9,6 @@
 libcairoboilerplate_la_LIBADD = $(top_builddir)/src/libcairo.la $(CAIRO_LDADD)
 
 if CAIRO_HAS_BEOS_SURFACE
-libcairoboilerplate_la_SOURCES += cairo-boilerplate-beos.cpp
 libcairoboilerplate_la_SOURCES += cairo-boilerplate-beos-private.h
 # BeOS system headers trigger this warning
 libcairoboilerplate_la_CXXFLAGS = -Wno-multichar
Index: cairo-1.4.14/src/Makefile.am
===================================================================
--- cairo-1.4.14.orig/src/Makefile.am	2008-04-14 15:08:25.000000000 +0100
+++ cairo-1.4.14/src/Makefile.am	2008-04-14 15:08:38.000000000 +0100
@@ -97,7 +97,6 @@
 libcairo_beos_sources =
 if CAIRO_HAS_BEOS_SURFACE
 libcairo_beos_headers = cairo-beos.h
-libcairo_beos_sources += cairo-beos-surface.cpp
 backend_pkgconfigs += cairo-beos.pc
 
 noinst_LTLIBRARIES = libcairo_beos.la


